pkg/controllers/nodeclaim/garbagecollection/controller.go

```diff
@@ -62,7 +62,7 @@ func (c *Controller) Reconcile(ctx context.Context) (reconcile.Result, error) {
 		return reconcile.Result{}, fmt.Errorf("listing cloudprovider machines, %w", err)
 	}
 	managedRetrieved := lo.Filter(retrieved, func(nc *karpv1.NodeClaim, _ int) bool {
-		return nc.Annotations[karpv1.ManagedByAnnotationKey] != "" && nc.DeletionTimestamp.IsZero()
+		return nc.Annotations[karpv1.NodePoolLabelKey] != "" && nc.DeletionTimestamp.IsZero()
```

Comment:
Are there cases where we get an instance back on a List() that isn't owned by Karpenter? Just thinking about it at a high-level, I think this check was here from when we did the migration, but I think now we would only return instances that were owned by Karpeneter anyways -- so this check should basically be a no-op

Reply:
Wouldn't we want the `nc.DeletionTimestamp.IsZero()` to be present so we don't attempt to garbage collect multiple times?